File: dumbwaiter_python/dumbwaiter/src/dumbwaiter/common/rpi.py
from enum import Enum

class PiPin(Enum):
  OpenerUp = 1
  OpenerDown = 2
  OpenerStop = 3
  Floor3Requested = 4
  Floor2Requested = 5
  Floor1Requested = 6
  StopRequested = 7
  AtFloor = 8


class RPi:
  def send_signal(self, pin):
    """send a signal on the specified pin"""
    pass

  def get_signal(self, pin):
    """ get the value from the specified pin """
    pass


class RPiDevice(RPi):
  def send_signal(self, pin):
    # TODO implement
    pass

  def get_signal(self, pin):
    # TODO implement
    return None

File: dumbwaiter_java/src/main/java/dumbwaiter/common/RPi.java
package dumbwaiter.common;


// RPi and interface to the RPi device
public interface RPi {
  void SendSignal(PiPin pin) throws RPiDeviceException; // send a signal on the target pin
  boolean GetSignal(PiPin pin) throws RPiDeviceException;   //  get a signal from a target pin, true when signal is on, false when off
}

